src/sentry/tasks/summaries/daily_summary.py

```diff
@@ -207,12 +207,37 @@ def build_summary_data(
                 group__in=([group for group in regressed_or_escalated_groups]),
                 type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_ESCALATING.value),
             )
-            if regressed_or_escalated_groups_today:
-                for activity in regressed_or_escalated_groups_today[:4]:
-                    if activity.type == ActivityType.SET_REGRESSION.value:
-                        project_ctx.regressed_today.append(activity.group)
-                    else:
-                        project_ctx.escalated_today.append(activity.group)
+
+            deduped_groups_by_activity_type: dict[ActivityType, set] = {
+                ActivityType.SET_REGRESSION: set(),
+                ActivityType.SET_ESCALATING: set(),
+            }
+
+            for activity in regressed_or_escalated_groups_today:
+                deduped_groups_by_activity_type.setdefault(ActivityType(activity.type), set()).add(
+                    activity.group
+                )
+
+                if activity.type == ActivityType.SET_ESCALATING.value:
+                    # if a group is already in the regressed set but we now see it in escalating, remove from regressed and add to escalating
+                    # this means the group regressed and then later escalated, and we only want to list it once
+                    if (
+                        activity.group
+                        in deduped_groups_by_activity_type[ActivityType.SET_REGRESSION]
+                    ):
+                        deduped_groups_by_activity_type[ActivityType.SET_REGRESSION].remove(
+                            activity.group
+                        )
+
+            for activity_type, groups in deduped_groups_by_activity_type.items():
+                for group in list(groups):
```

Comment:
I think you can slice the list and add it once like before instead of iterating

Reply:
oops you're right, was reading it wrong